### PR TITLE
Usage report of API and S3 gateway

### PR DESCRIPTION
**Pull request by nopcoder**
- not enabled by default
- store information into kv database
- endpoint support json and plain text report

Close https://github.com/treeverse/lakeFS/issues/7280

### CODE REVIEW

File: pkg/api/controller.go [hunk 1]
```diff
@@ -5193,3 +5198,47 @@ func extractLakeFSMetadata(header http.Header) map[string]string {
 	}
 	return meta
 }
+
+func (c *Controller) GetUsageReportSummary(w http.ResponseWriter, r *http.Request) {
```
Comment (idanovo) [inline pkg/api/controller.go:5202]: No RBAC level protection? Any user who can log in to the installation will get an answer for this API call?

**Review by idanovo — CHANGES_REQUESTED** [review]
Not sure I understand who can make this API call

File: pkg/api/controller.go [hunk 2]
```diff
@@ -5193,3 +5198,47 @@ func extractLakeFSMetadata(header http.Header) map[string]string {
...
+	ctx := r.Context()
+
+	// c.LogAction(ctx, "get_usage_report_summary", r, "", "", "")
```
[the remark this answers is not in the document]
Comment (nopcoder) [reply pkg/api/controller.go:5205]: Will remove - thanks!

File: cmd/lakefs/cmd/run.go [hunk 3]
```diff
@@ -174,6 +174,14 @@ var runCmd = &cobra.Command{
 		}
 		defer func() { _ = c.Close() }()
 
+		// usage report setup - nil if disabled
```
[the remark this answers is not in the document]
Comment (nopcoder) [reply cmd/lakefs/cmd/run.go:177]: Updated the code - the default reporter is no op.

File: pkg/api/controller.go (hunk 1 above)
Comment (nopcoder) [reply pkg/api/controller.go:5202]: Should be available to any signed-in user. No special permissions required.

Comment (idanovo) [reply pkg/api/controller.go:5202]: @ozkatz @talSofer WDYT?

**Review by idanovo — APPROVED** [review]
I'm still not sure if it make sense that every user can make this report but it's a product decision I guess.

**Comment by nopcoder** [comment]
> I'm still not sure if it make sense that every user can make this report but it's a product decision I guess.

Stated the product decision

**Pull request merged**